backend/src/ee/services/pam-resource/aws-iam/aws-iam-federation.ts

```diff
@@ -0,0 +1,209 @@
+};
+
+export const validatePamRoleConnection = async (
+  connectionDetails: TAwsIamResourceConnectionDetails,
+  projectId: string
+): Promise<boolean> => {
+  const stsClient = createStsClient();
+
+  try {
+    await stsClient.send(
+      new AssumeRoleCommand({
+        RoleArn: connectionDetails.roleArn,
+        RoleSessionName: `infisical-pam-validation-${Date.now()}`,
+        DurationSeconds: AWS_STS_MIN_DURATION_SECONDS,
+        ExternalId: projectId
+      })
+    );
+
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const validateTargetRoleAssumption = async ({
+  connectionDetails,
+  targetRoleArn,
+  projectId
+}: {
+  connectionDetails: TAwsIamResourceConnectionDetails;
+  targetRoleArn: string;
+  projectId: string;
+}): Promise<boolean> => {
+  const stsClient = createStsClient();
+
+  try {
+    // First assume the PAM role
+    const pamRoleCredentials = await stsClient.send(
+      new AssumeRoleCommand({
+        RoleArn: connectionDetails.roleArn,
+        RoleSessionName: `infisical-pam-validation-${Date.now()}`,
+        DurationSeconds: AWS_STS_MIN_DURATION_SECONDS,
+        ExternalId: projectId
+      })
+    );
+
+    if (!pamRoleCredentials.Credentials) {
+      return false;
+    }
+
+    // Then use the PAM role credentials to assume the target role
+    const pamStsClient = new STSClient({
+      region: AWS_STS_DEFAULT_REGION,
+      useFipsEndpoint: crypto.isFipsModeEnabled(),
+      sha256: CustomAWSHasher,
+      credentials: {
+        accessKeyId: pamRoleCredentials.Credentials.AccessKeyId!,
+        secretAccessKey: pamRoleCredentials.Credentials.SecretAccessKey!,
+        sessionToken: pamRoleCredentials.Credentials.SessionToken
+      }
+    });
+
+    await pamStsClient.send(
+      new AssumeRoleCommand({
+        RoleArn: targetRoleArn,
+        RoleSessionName: `infisical-pam-target-validation-${Date.now()}`,
+        DurationSeconds: AWS_STS_MIN_DURATION_SECONDS,
+        ExternalId: projectId
+      })
+    );
+
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+/**
+ * Assumes the target role and generates a federated console sign-in URL.
+ */
+export const generateConsoleFederationUrl = async ({
+  connectionDetails,
+  targetRoleArn,
+  roleSessionName,
+  projectId,
+  sessionDuration
+}: {
+  connectionDetails: TAwsIamResourceConnectionDetails;
+  targetRoleArn: string;
+  roleSessionName: string;
+  projectId: string;
+  sessionDuration: number;
+}): Promise<{ consoleUrl: string; expiresAt: Date }> => {
+  const stsClient = createStsClient();
+
+  // First assume the PAM role
+  const pamRoleCredentials = await stsClient.send(
+    new AssumeRoleCommand({
+      RoleArn: connectionDetails.roleArn,
+      RoleSessionName: `infisical-pam-${Date.now()}`,
+      DurationSeconds: sessionDuration,
+      ExternalId: projectId
+    })
+  );
+
+  if (!pamRoleCredentials.Credentials) {
+    throw new Error("Failed to assume PAM role");
+  }
```

Comment:
let's throw BadRequest or InternalServer error here? I think if we throw errors like this, it won't propagate neatly to the UI